pulsar-common/src/main/java/org/apache/pulsar/common/util/collections/ConcurrentLongLongPairHashMap.java

```diff
@@ -322,7 +325,10 @@ private static final class Section extends StampedLock {
         LongPair get(long key1, long key2, int keyHash) {
             long stamp = tryOptimisticRead();
             boolean acquiredLock = false;
-            int bucket = signSafeMod(keyHash, capacity);
+            // add local variable here, so OutOfBound won't happen
+            long[] table = this.table;
+            // calculate table.length / 4 as capacity to avoid rehash changing capacity
+            int bucket = signSafeMod(keyHash, table.length / ITEM_SIZE);
```

Comment:
Btw, it seems that the original code has a bug in calculating the index in `table` for the bucket. I think that it's a a bug in the original code. It seems that there should be a separate `bucketIndex` variable which is calculated by `bucket * ITEM_SIZE`. @merlimat WDYT?

Reply:
I'll create a separate PR to fix that issue. The main impact is non-optimal lookup since hash collisions will happen when they shouldn't.

Reply:
I created #21107  to fix the bugs in the hash bucket algorithm implementation.